sapl/rules/apps.py

```diff
@@ -242,20 +242,93 @@ def cria_usuarios_padrao():
     rules.cria_usuarios_padrao()
 
 
+def reset_id_model(model):
+
+    # - pg_get_serial_sequence recupera o nome da sequencia que foi
+    # associada para o campo 'id' de table_name, independente do padrão
+    # de criação do nome.
+    # - setval registra:
+    #   - coalesce(max("id"), 1) - o id max para a chave retornada ou 1,
+    #   - max("id") IS NOT null - true se max id retorna nulo
+    # - retornos:
+    #    - [(max id, )]
+    #    - [(None, )] caso não exista sequencia que vincule
+    #                 à tabela e ao id da tabela,
+    #                 mesmo que sequencia exista mas sem vínvulo.
+
+    query = """SELECT setval(pg_get_serial_sequence('%(table_name)s','id'),
+                coalesce(max("id"), 1), max("id") IS NOT null) 
+                FROM "%(table_name)s";
+            """ % {
+        'table_name': model._meta.db_table
+    }
```

Comment:
```suggestion
query = """SELECT tc.table_name
           FROM information_schema.columns tc
           JOIN pg_tables t ON (tc.table_name = t.tablename)
           WHERE tc.column_name = 'id' AND tc.column_default IS NULL
           AND t.schemaname='public'""
```
 
A consulta abaixo vai te trazer todas as tabelas que não possuem `DEFAULT` (no caso a sequence) vinculada a ela. Poderíamos iterar sobre essa lista. O que acha?

Reply:
bom... temos aí duas abordagens diferentes...

- a inicial, através da execução trivial do post_migrate, dado um app, para cada model seu, força-se o próximo ID ser o max_id + 1 dado o sequence vínculado se este existir. Caso não exista sequence(`if rows[0][0] is None`) ou a tabela não tenha ID , levando em consideração que, pela características gerais do sapl, um usuário/admin e/ou função qualquer de manutenção/migração, não deveria alterar os sequences, dentro do IF, então: 1) é excluído qualquer sequence que possua o nome que deveria ser do model em questão associado a esse model ou a qualquer outro (cascade do drop); 2)  cria esse sequence com start 1 vinculando ao model e ao campo (isso não da erro nesse momento e não dará depois já que o passo 4 ainda será executado); 3) altera o campo do model definindo seu default (vínculos e defaults são coisas diferentes); e 4) redefine o sequence para max_id+1.

- a segunda abordagem, não é feita uma revisão geral dos IDs e seus sequences, não condiz com o esquema do post_migrate que é ativado n vezes onde n é o número de apps... então, nessa abordagem, o post_migrate deveria chamar a função reset_id_model (daí um nome melhor seria cria_sequences_inexistentes), mas o post_migrate é chamado n vezes, dessa forma, todas as n execuções deveriam ser descartadas e somente só acontecer se, digamos, app_config.name = 'sapl.rules', daí chamaria a função cria_sequences_inexistentes que recuperaria todas as tabelas sem default e 1) criaria o sequence com start 1 (ou recupera o max_id+1 previamente e já cria com ele) vinculando ao tabela e a seu id; 2) altera o default; e 3) se no passo 1, o start foi 1 então, redefine-se o max_id+1